libs/wingsdk/src/ex/react-app.md

```diff
@@ -0,0 +1,88 @@
+---
+title: React Website
+id: react-website
+description: A built-in resource for creating deployable websites using the React framework.
+keywords: [Website, React, deployment, build]
+sidebar_position: 1
+---
+
+The `ex.ReactApp` resource represents a website, built using React, that can be both hosted in the cloud or to run on a development hot-reload server in the sim target.
+
+## Usage
+
+### Initialization
+
+Using the default arguments:
+
+```ts
+bring ex;
+bring util;
+
+let website = new ex.ReactApp(projectPath: "./client");
+```
+
+or customizing them:
+
+```ts
+bring ex;
+bring util;
+
+let website = new ex.ReactApp(
+  projectPath: "./client",
+  isDevRun: util.tryEnv("ENV") == "dev" // `true`` by default. will run the start command if true, and the build command if not
+  buildDir: "/dist" // default is "/build"
+  startCommand: "pnpm start" // default is "npm start"
+  buildCommand: "pnpm build" // default is "npm build"
+  hostProps: {} // properties that apply to the react app host, which is a `cloud.Website` resource
+  localPort: "4000" // default is 3001
+ );
+```
+
+- When calling `wing it` with `isDevRun` set to `true`, Wing will start both the simulator and React's development server.
+- When calling `wing it` or `wing compile` for the `sim` target with `isDevRun` set to `false`, Wing will build React and serve it via the [Website resource](../01-cloud/website.md).
+- When calling `wing compile` for any other target, Wing will build React locally and generate output files, preparing it for deployment to the cloud using the [Website resource](../01-cloud/website.md).
+
+#### using wing variables within react code
+
+`ex.ReactApp` allows you to pass preflight arguments from wing to the React app using `addEnvironmentVariable` method:
```

Comment:
> @staycoolcall911 I don't have a strong preference between these names (maybe @eladb has an opinion?), but the name should match the corresponding method in cloud.Function since it has the same functionality.

@Chriscbr - thanks for the feedback, I wasn't aware of the `cloud.Function` method. In that case, let's keep it consistent. Sorry for the back and forth @tsuf239... let's revert to use `addEnvironment`